base/strings/search.jl

```diff
@@ -572,8 +325,39 @@ julia> findprev('o', "Hello to the world", 18)
 15
 ```
 """
-findprev(ch::AbstractChar, string::AbstractString, start::Integer) =
-    findprev(==(ch), string, start)
+findprev(a::AbstractChar, b::AbstractString, start::Integer) = findprev(isequal(a), b, start)
+
+function findprev(p::Function, b::AbstractString, stop::Integer)
+    i = Int(stop)
+    first = firstindex(b)
+    i < first && return nothing
+    i > lastindex(b) && throw(BoundsError(b, i))
+    i = thisind(b, i)
+    @inbounds while i ≥ first
+        p(b[i]) && return i
+        i = prevind(b, i)
+    end
+    return nothing
+end
+
+function findprev(p::Fix2Eq{<:AbstractChar}, b::Union{String,SubString{String}}, stop::Integer)
+    i = Int(stop)
+    first = firstindex(b)
+    i < first && return nothing
+    n = ncodeunits(b)
+    i > n && throw(BoundsError(b, i))
+    i = thisind(b, i)
+    probe = first_utf8_byte(p.x)
+    while true
+        offset = search_backward(probe, codeunits(b), n - i)
+        if offset < 0
+            return nothing
+        elseif p(b[offset+first])
+            return offset + first
+        end
+        i = prevind(b, i)
+    end
```

Comment:
Maybe we want to make this explicit ?
```suggestion
    end
    return nothing
```

Reply:
Thank you. I think that `return` is rather confusing because it will never be executed considering that the while loop above neither finishes nor breaks.

Reply:
I see. Thanks for the explanation.